packages/@lwc/engine-dom/src/patches/global-registry.ts

```diff
@@ -0,0 +1,451 @@
+                return nativeGet.call(nativeRegistry, tagName)!;
+            }
+            return NativeCtor;
+        });
+    };
+
+    // This constructor is invoked when we call `new instanceDefinition.UserCtor()`
+    // @ts-ignore
+    window.HTMLElement = function HTMLElement(this: HTMLElement) {
+        // Upgrading case: the pivoting class constructor was run by the browser's
+        // native custom elements and we're in the process of running the
+        // "constructor-call trick" on the natively constructed instance, so just
+        // return that here.
+        // This code path is also called when LWC `new`s a PivotCtor.
+        const instance = upgradingInstance;
+        if (!isUndefined(instance)) {
+            upgradingInstance = undefined;
+            return instance;
+        }
+        // Construction case: we need to construct the pivoting instance and return it.
+        // This is possible when the user register it via global registry and instantiate
+        // it via `new Ctor()`.
+        const { constructor } = this;
+        const definition = globalDefinitionsByClass.get(constructor as CustomElementConstructor);
+        if (isUndefined(definition) || isUndefined(definition.PivotCtor)) {
+            // This code path is hit if someone `new`s a class that extends `HTMLElement` without
+            // doing `customElements.define()` first. This matches native browser behavior:
+            // https://stackoverflow.com/a/61883392
+            throw new TypeError('Illegal constructor');
+        }
+        // This constructor is ONLY invoked when it is the user instantiating
+        // an element via new Ctor while Ctor is a registered global constructor.
+        const { PivotCtor, UserCtor } = definition;
+        return new PivotCtor(UserCtor);
+    };
+    HTMLElement.prototype = NativeHTMLElement.prototype;
+
+    // This method patches the DOM and returns a helper function for LWC to register names and associate
+    // them to a constructor while returning the pivot constructor, ready to instantiate via `new`.
+    createPivotConstructor = (tagName: string, constructor: CustomElementConstructor) => {
+        tagName = tagName.toLowerCase();
+        let PivotCtor = pivotCtorByTag.get(tagName);
+        if (isUndefined(PivotCtor)) {
+            const definition = getOrCreateDefinitionForConstructor(constructor);
+            PivotCtor = createPivotingClass(tagName, definition);
+            // Register a pivoting class as a global custom element
+            nativeDefine.call(nativeRegistry, tagName, PivotCtor);
+            definition.PivotCtor = PivotCtor;
+            // Only cache after nativeDefine has been called, because if it throws an error
+            // (e.g. for an invalid tag name), then we don't want to cache anything.
+            definitionForConstructor.set(constructor, definition);
+            pivotCtorByTag.set(tagName, PivotCtor);
+        }
+        return PivotCtor;
+    };
```

Comment:
Simplified by removing the `createScopedRegistry`. Now we just export a function instead of a function that returns a function.

This function could probably be simplified further by just taking a `tagName` as an argument (we create the `constructor` from scratch every time anyway), but I need to think through the implications a bit more if we were to expose this as a third-party package.